2017 Robot/src/org/usfirst/frc/team4999/lights/Packet.java

```diff
@@ -0,0 +1,162 @@
+package org.usfirst.frc.team4999.lights;
+
+import java.nio.ByteBuffer;
+
+/**
+ * A command sent to the arduino to light up a pixel or area of pixels with a certain color.
+ * Follows <a href="https://github.com/momentumfrc/2017Steamworks/wiki/Light-strings-on-the-Robot">this</a> structure.
+ * @author jordan
+ *
+ */
+public class Packet {
+	// The max size of a packet is 16 bytes
+	// Before a packet is sent, B_BUFFER is filled with that packet's data. B_BUFFER is then passed to the I2C writeBulk() method
+	private static final ByteBuffer B_BUFFER = ByteBuffer.allocateDirect(16);
+	
+	// All according to specification
+	private final byte DISPLAY_FRAME = 0x01;
+	private final byte SET_SINGLE = 0x02;
+	private final byte SET_AREA = 0x03;
```

Comment:
AREA infers a 2-dimensional thing.  A better choice would be SERIES or RUN.  I personally prefer RUN since we used that all the time in our imaging code at Xerox.  You'll also find that term used in Run Length encoding, an early compression technique where long sequences of identical bytes are replaced with a value/count pair.